### ISSUE TITLE
Accessibility testing: TalkBack findings

### ISSUE THREAD

**Issue by devinreams** [mozilla-lockwise/lockwise-android#223]
Related to #44 

I ran through all the app screens and elements currently in place and documented those all below.

A few areas to highlight:

1. FxA web content is a bit inconsistently read and keyboards appear vs not (may be web content related not app)

2. The search/filter experience on the list view is not quite polished and as what I see in Android’s own interfaces (Settings screen was a good compare/contrast)

3. The Settings screen elements seems a little disconnected from each other so not as clearly connected/usable as possible

And some findings from the other accessbility review for content labeling #200 show up clearly here (reuse of labels or missing labels).

I will file a follow-up PR to document the test plan (in addition to whats already in docs).

- - -

The reading of on-screen interface items and objects. Testing for:
- All on-screen navigation and button titles are read
- All on-screen alerts and popovers (interrupting the interface) are read
- All labels and text elements on screen are read
- Interaction elements like input boxes and filters have meaningful instructions
- Links to open websites are clearly indicated and read
- Swiping left/right to select next/previous elements works usefully
- Interface scrolling is available where needed

### Welcome Screen (first run)

- App Name: “Lockbox Firefox Lockbox” is read :warning: (repetitive but screen is still being worke on)
- Tagline: “Take your passwords everywhere” is read
- “Get Started button” is read — “double tap to activate”
- “To use Firefox Lockbox...” is read
- “Learn more button” is read — “double tap to activate”

### Sign in to Firefox Accounts (web view: email)

- When opened, no default interface item is selected/read :warning:
- Button “X” to close is “unlabeled button” :warning:
- “Enter your email to continue to Firefox Lockbox...” is read
- “Edit box - Email” is read
- Keyboard does not appear when input selected :warning: (had to hit “Continue” for it to appear, but works OK on “Password” input..)
- “Valid email required” error is read
- “Editing edit box: Email” is read
- “Continue” is read (:warning: not described as a button)

### Sign in to Firefox Accounts (web view: enter password)

- When loaded, interface is selected and read
- “Password edit box: Password” is read
- Double tap on “Password” does show keyboard (unlike email, note above)
- “Keyboard hidden” is read upon submit

### Sign in to Firefox Accounts (web view: enter password second run)

- When loaded interface is not selected and read
- Double tap on “Password” does not show keyboard (unlike above) :warning:

### Sign in to Firefox Accounts (web view: confirm this sign in)

- When loaded, interface is not selected nor read :warning:
- When confirmed, interface is not selected nor read, transition is not announced :warning:
- “Continue button” is read — “double tap to activate”

### List View

- When loaded, “List” is selected and read :warning: (first item could probably be the “Lockbox All Entries A to Z” filter instead)
- Drawer is “unlabeled button” :warning:
- “All Entries A to Z” is read :warning: (not announced its a filter)
- “Alphabetically” is read - “double tap to activate” :warning” (doesn’t match label of “All Entries A to Z”? not useful to explain it will filter?)
- Double tap “Alphabetically” and no transition or explanation its already selected/used :warning:
- Unclear if possible to tap or close the popup filter :warning:
- “All Entries Recent” is read :warning: (useful title for the filtered view?)
- Search field announced as “button” :warning: (not descriptive)
- “Search your entries edit box” is read and keyboard activated
- “Button search clear text button” :warning: (clear button description is complex)
- Each list item can be navigated using left/right swipes
- Each list item is read with website and username
- “Double tap to activate” is read on each item
- When returning from detail view, filtered text edit box is read but the context (like “accounts”) is not read :warning:

## Detail view

- When loaded the web address (https://) is selected and read first :warning: (could be the nav bar title instead)
- Back arrow is “unlabeled button” :warning:
- Web address is read: “https://accounts.firefox.com edit box web address”
- Username is read: “faketester3 edit box username”
- Double tap on field does not copy to clipboard :warning: (expected?)
- Double tap copy buttons are read
- “Username copied” and “Password copied” alerts are read
- Password is read: “password 21 characters edit box password”
- Web address, Username, and Password, including revealed, all say “edit box” and “double tap to enter text” :warning: (fields not actually editable)
- “Display your password in plaintext button” is read
- Double tap reveal button and password field is not selected/re-read nor anything announced :warning:
- After revealed, “display your password in plaintext button” label does not change to opposite meaning :warning: (hide)
- “Learn how to edit this entry” link is read

## Navigation drawer

- Once loaded, “Menu, Firefox Lockbox” is selected and read
- All list items can be navigated with left/right gesture
- “double tap to activate” is read for each labeled item

## Welcome Screen (locked)

- Once loaded, nothing is selected nor read :warning:
- “Unlock Firefox Lockbox” button is read
- Android PIN/fingerprint screens are read

## Account Screen

- Once loaded, back button (not nav bar title) is selected and read :warning:
- Back “unlabeled” button :warning:
- “Firefox Account profile image” is read
- “Lockbox Tester” (real name) is read
- Text on screen is read
- Disconnect button “double tap to activate” is read
- Popover text is read
- Button to cancel and disconnect is read

## Settings Screen

- Once loaded, back button (not nav bar title) is selected and read :warning:
- “Security..” heading is read (but says 7 items because it counts Unlock + Setting + Autofill + Setting)
- “Unlock with fingerprint” is read
- Unlock “off switch” setting is not specifically announced as being for that setting :warning:
- “Auto Lock 5 minutes” (setting and value) is read together
- No announcement to change the auto lock setting :warning:  (feature not in place yet)
- No ability to double tap and change auto lock setting :warning: (feature not in place yet)
- “Autofill.. Let Firefox ...” setting is read
- Autofill “off switch” setting is not specifically announced as being for that setting :warning:
- “Send usage data... Mozilla strives” is read
- Usage data “on switch, double tap to toggle” is not specifically announced as being for that setting :warning:
- “Learn more button” is next after the usage switch so disconnected from the paragraph its related to :warning:
- “App Version” is read

**Comment by devinreams** [comment]
Stale. Closing in favor of #800
